src/react/actions/replication.js

```diff
@@ -47,9 +53,12 @@ export function saveReplication(
 ): ThunkStatePromisedAction {
   return (dispatch, getState) => {
     const state = getState();
+    const { oidc } = getState();
     const { managementClient, instanceId } = getClients(state);
     const accountId = getAccountId(state);
     dispatch(networkStart('Creating replication'));
+    replication.roleName = roleName;
+    replication.RoleSessionName = (oidc.user) ? oidc.user.profile.email : '';
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Tbh at this point a functhion might be better... :smirk:

Reply:
Yes, many thanks for the feedback!

Reply:
As the roleSessionName is now computed by pensieve-api dynamically, I removed most of these lines: https://github.com/scality/pensieve-api/pull/143

I created a new function anyway, as it canmay be helpful in the future: https://github.com/scality/zenko-ui/pull/203/files#diff-ff376b58fc91513c886b2a8892cd7f7f1303edc5b9be29248b666d2a5d0c8985R43

Reply:
We use this function here: https://github.com/scality/zenko-ui/pull/203/files#diff-e728917123f674a0661856cb2a62fe471f3bbdc2435af7063dfdb4b961f7b15aR21